satellite/satellitedb/storagenodeaccounting.go

```diff
@@ -200,21 +200,41 @@ func (db *StoragenodeAccounting) QueryPaymentInfo(ctx context.Context, start tim
 func (db *StoragenodeAccounting) QueryStorageNodeUsage(ctx context.Context, nodeID storj.NodeID, start time.Time, end time.Time) (_ []accounting.StorageNodeUsage, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	query := `SELECT at_rest_total, start_time, 
-		LAG(at_rest_total) OVER win AS prev_at_rest, 
-		LAG(start_time) OVER win AS prev_start_time
-		FROM accounting_rollups
-		WHERE id IN (
-			SELECT MAX(id)
-			FROM accounting_rollups
-			WHERE node_id = ?
-			AND ? <= start_time AND start_time <= ?
-			GROUP BY start_time
-			ORDER BY start_time ASC
-		)
-		WINDOW win AS (ORDER BY start_time)`
+	lastRollup, err := db.db.Find_AccountingTimestamps_Value_By_Name(ctx, dbx.AccountingTimestamps_Name(accounting.LastRollup))
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+	if lastRollup == nil {
+		return nil, Error.New("unable to find last accounting rollup timestamp")
+	}
+
+	start, end = start.UTC(), end.UTC()
+
+	query := `SELECT SUM(at_rest_total), start_time
+				FROM (
+					SELECT r.at_rest_total, r.start_time
+					FROM (
+						SELECT MAX(id) as id
+						FROM accounting_rollups
+						WHERE node_id = ?
+						AND ? <= start_time AND start_time <= ?
+						GROUP BY start_time
```

Comment:
If I'm not mistaken this subquery only returns one row, one `id`.
If that's the case I would avoid the `GROUP BY`

```sql
SELECT id
FROM accounting_rollups
WHERE node_id = ?
AND ? <= start_time AND start_time <= ?
LIMIT 1
```

Reply:
I didn't know that `start_time` was unique, but it's filtering by `node_id` so having a maximum of a single value is that same value.

Reply:
@ifraixedes, @cam-a  `MAX` is leftover from debugging when I removed code that deletes today`s `rollups`. I left it because in our DB schema for `accounting` we allow duplicated values for the same timestamp